File: 3577-countTheNumberOfComputerUnlockingPermutations/countPermutations.go
package main

const MOD = 1000000007

func permutations(n int) int {
	var ans int64 = 1
	for i := 2; i <= n; i++ {
		ans = (ans * int64(i)) % MOD
	}
	return int(ans)
}

func countPermutations(complexity []int) int {
	min := complexity[0]
	n := len(complexity)
	for i := 1; i < n; i++ {
		if complexity[i] < min {
			return 0
		}
	}
	return permutations(n - 1)
}

File: 3577-countTheNumberOfComputerUnlockingPermutations/countPermutations.c
#define MOD 1000000007
#define bool char
#define true 1
#define false 0

inline bool possible(int base, int *nums, int numsSize) {
  for (int i = 0; i < numsSize; i++) {
    if (nums[i] <= base) {
      return false;
    }
  }
  return true;
}

inline int permutations(int n) {
  // I's basically a factorial
  // n * n-1 * n-2...
  long long res = 1;
  for (int i = 2; i <= n; i++) {
    res = (res * i) % MOD;
  }
  return (int)res;
}

int countPermutations(int *complexity, int complexitySize) {
  if (possible(complexity[0], &complexity[1], complexitySize - 1)) {
    return permutations(complexitySize - 1);
  } else {
    return 0;
  }
}
